app/User.php

```diff
@@ -326,4 +327,116 @@ public function bookmarked($model)
             ->withTimestamps();
     }
 
-}
+    public function delegations()
+    {
+        return $this->hasMany(UserDelegation::class, 'delegator_user_id');
+    }
+
+    /**
+     * User delegates. Many-to-many relationship.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function delegates()
+    {
+        return $this->belongsToMany('App\User', 'user_delegations', 'delegator_user_id', 'delegate_user_id')
+                    ->using(UserDelegation::class)
+                    ->withPivot('starting', 'until', 'gets_reminders')
+                    ->withTimestamps();
+    }
+
+    /**
+     * Current user delegates.
+     * 
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function currentDelegates()
+    {
+        return $this->delegates()
+            ->where('user_delegations.starting', '<=', DB::raw('now()'))
+            ->where(function($q) {
+                $q->where('user_delegations.until', '>', DB::raw('now()'));
+                $q->orWhereNull('user_delegations.until');
+            });
+    }
+
+    /**
+     * User's current delegates that get reminders.
+     *
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    public function currentReminderDelegates()
+    {
+        return $this->currentDelegates()->where('gets_reminders', '=', true);
+
+    }
+
+    /**
+     * Email addresses of the user's current delegates that get reminders.
+     *
+     * @return array
+     */
+    public function currentReminderDelegateEmailAddresses()
+    {
+        return $this->currentReminderDelegates()->pluck('email')->all();
+    }
+
+    /**
+     * User delegators. Inverse of delegate many-to-many relationship.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function delegators()
+    {
+        return $this->belongsToMany('App\User', 'user_delegations', 'delegate_user_id', 'delegator_user_id')
+                    ->using(UserDelegation::class)
+                    ->withPivot('starting', 'until', 'gets_reminders')
+                    ->withTimestamps();
+    }
+
+    /**
+     * Current user delegators.
+     * 
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function currentDelegators()
+    {
+        return $this->delegators()
+            ->where('user_delegations.starting', '<=', DB::raw('now()'))
```

Comment:
Maybe change to `->whereDate('user_delegations.starting', '<=', now())` and similar on line 407.

Reply:
@wunc This correction was applied in next commit.